index.js

```diff
@@ -25,61 +25,61 @@ instance.prototype.updateConfig = function(config) {
 	self.init_tcp();
 };
 
+instance.prototype.init = function() {
+	var self = this;
+
+	debug = self.debug;
+	log = self.log;
+
+	self.init_tcp();
+};
+
 instance.prototype.incomingData = function(data) {
 	var self = this;
 	debug(data);
 
 	// Match part of the copyright response from unit when a connection is made.
-	// Send Info request which should reply with Matrix setup, eg: "V8X4 A8X4"
 	if (self.login === false && data.match("Extron Electronics")) {
 		self.status(self.STATUS_WARNING,'Logging in');
-		self.socket.write("I"+ "\n");
+		self.socket.write("I"+ "\n"); // Matrix information request
 	}
 
 	if (self.login === false && data.match("Password:")) {
 		self.status(self.STATUS_WARNING,'Logging in');
-		self.socket.write(""+ "\n");
+		self.socket.write("\r" +self.config.password+ "\r"); // Enter Password Set
 	}
 
-	// Match first letter of expected response from unit.
-	else if (self.login === false && data.match(/V|60/)) {
+	// Match login sucess response from unit.
+	else if (self.login === false && data.match("Login")) {
```

Comment:
match() are usually meant to be used with regular expressions, not strings. Better to write /Login/, for clarity. (again, nitpicking, and personal feelings ;P )

Reply:
Understood